diff --git a/CMakeLists.txt b/CMakeLists.txt
index eab7e01..bcc1026 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,6 +50,13 @@ if( NOT DEFINED IOT_PLATFORM_NAME )
         option( IOT_NETWORK_USE_OPENSSL
                 "Set this to ON to use a network abstraction implemented on OpenSSL. When OFF, the mbed TLS network abstraction is used."
                 OFF )
+    elseif( ${CMAKE_SYSTEM_NAME} STREQUAL "VxWorks" )
+        set( IOT_PLATFORM_NAME "posix" CACHE STRING "Port to use for building the SDK." )
+
+        # Provide an option to use the OpenSSL network abstraction on VxWorks.
+        option( IOT_NETWORK_USE_OPENSSL
+                "Set this to ON to use a network abstraction implemented on OpenSSL. When OFF, the mbed TLS network abstraction is used."
+                OFF )
     elseif( ${CMAKE_SYSTEM_NAME} STREQUAL "Windows" )
         set( IOT_PLATFORM_NAME "win32" CACHE STRING "Port to use for building the SDK." )
 
diff --git a/platform/ports/posix/CMakeLists.txt b/platform/ports//posix/CMakeLists.txt
index bf54445..fbf3219 100644
--- a/platform/ports/posix/CMakeLists.txt
+++ b/platform/ports/posix/CMakeLists.txt
@@ -2,13 +2,6 @@ include( CheckCCompilerFlag )
 include( CheckTypeSize )
 include( CheckFunctionExists )
 
-# Check that the -lrt flag works.
-check_c_compiler_flag( -lrt HAS_C_FLAG_lrt )
-
-if( NOT HAS_C_FLAG_lrt )
-    message( FATAL_ERROR "Compiler flag -lrt must be supported." )
-endif()
-
 # Check for POSIX threads.
 find_package( Threads REQUIRED )
 
@@ -34,7 +27,7 @@ foreach( POSIX_TYPE ${REQUIRED_POSIX_TYPES} )
 endforeach()
 
 # Check for some required POSIX functions. This is not intended to be a comprehensive list.
-set( CMAKE_REQUIRED_LIBRARIES rt Threads::Threads )
+set( CMAKE_REQUIRED_LIBRARIES Threads::Threads )
 list( APPEND REQUIRED_POSIX_FUNCTIONS
       clock_gettime time localtime_r strftime timer_create timer_delete
       timer_settime pthread_create pthread_attr_init pthread_attr_setdetachstate
@@ -90,7 +75,7 @@ add_library( iotplatform
                 ${PLATFORM_SOURCES} )

 # Set the dependencies of this platform layer.
-target_link_libraries( iotplatform INTERFACE Threads::Threads rt ${TLS_LIBRARY_LINKER_FLAG} )
+target_link_libraries( iotplatform INTERFACE Threads::Threads ${TLS_LIBRARY_LINKER_FLAG} )
 
 # Set platform sources in the parent scope for directory organization.
 set( PLATFORM_SOURCES ${PLATFORM_SOURCES} PARENT_SCOPE )
diff --git a/platform/ports/posix/source/iot_network_openssl.c b/platform/ports/posix/source/iot_network_openssl.c
index da804ef..44d6af5 100644
--- a/platform/ports/posix/source/iot_network_openssl.c
+++ b/platform/ports/posix/source/iot_network_openssl.c
@@ -36,7 +36,7 @@
 
 /* POSIX includes. */
 #include <errno.h>
-#include <poll.h>
+#include <sys/poll.h>
 #include <pthread.h>
 #include <signal.h>
 #include <sys/ioctl.h>
@@ -45,6 +45,8 @@
 #include <arpa/inet.h>
 #include <netdb.h>
 #include <sys/socket.h>
+#include <netinet/in.h>
+#include <netinet/ip.h>
 
 /* OpenSSL includes. */
 #include <openssl/err.h>
