diff --git a/CMakeLists.txt b/CMakeLists.txt
index 409d092..1aa9429 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,6 +29,7 @@ option( IOT_BUILD_CLONE_SUBMODULES
 
 # Check for system support.
 list( APPEND SUPPORTED_SYSTEMS "Linux" )
+list( APPEND SUPPORTED_SYSTEMS "VxWorks" )
 
 if( NOT ${CMAKE_SYSTEM_NAME} IN_LIST SUPPORTED_SYSTEMS )
     message( FATAL_ERROR "Unsupported system: ${CMAKE_SYSTEM_NAME}. Currently supported systems are: ${SUPPORTED_SYSTEMS}." )
@@ -77,6 +78,10 @@ if( ${CMAKE_SYSTEM_NAME} STREQUAL "Linux" )
     add_definitions( -DIOT_SYSTEM_TYPES_FILE="posix/iot_platform_types_posix.h" )
     add_subdirectory( platform/source/posix )
 endif()
+if( ${CMAKE_SYSTEM_NAME} STREQUAL "VxWorks" )
+    add_definitions( -DIOT_SYSTEM_TYPES_FILE="posix/iot_platform_types_posix.h" )
+    add_subdirectory( platform/source/posix )
+endif()
 
 # Common libraries (linear containers, logging, etc.)
 add_subdirectory( lib/source/common )
@@ -100,6 +105,9 @@ add_subdirectory( third_party/tinycbor )
 if( ${CMAKE_SYSTEM_NAME} STREQUAL "Linux" )
     add_subdirectory( demos/app/posix )
 endif()
+if( ${CMAKE_SYSTEM_NAME} STREQUAL "VxWorks" )
+    add_subdirectory( demos/app/posix )
+endif()
 
 # Test executables.
 if( ${IOT_BUILD_TESTS} )
diff --git a/platform/source/posix/CMakeLists.txt b/platform/source/posix/CMakeLists.txt
index bf54445..fbf3219 100644
--- a/platform/source/posix/CMakeLists.txt
+++ b/platform/source/posix/CMakeLists.txt
@@ -2,13 +2,6 @@ include( CheckCCompilerFlag )
 include( CheckTypeSize )
 include( CheckFunctionExists )
 
-# Check that the -lrt flag works.
-check_c_compiler_flag( -lrt HAS_C_FLAG_lrt )
-
-if( NOT HAS_C_FLAG_lrt )
-    message( FATAL_ERROR "Compiler flag -lrt must be supported." )
-endif()
-
 # Check for POSIX threads.
 find_package( Threads REQUIRED )
 
@@ -34,7 +27,7 @@ foreach( POSIX_TYPE ${REQUIRED_POSIX_TYPES} )
 endforeach()
 
 # Check for some required POSIX functions. This is not intended to be a comprehensive list.
-set( CMAKE_REQUIRED_LIBRARIES rt Threads::Threads )
+set( CMAKE_REQUIRED_LIBRARIES Threads::Threads )
 list( APPEND REQUIRED_POSIX_FUNCTIONS
       clock_gettime time localtime_r strftime timer_create timer_delete
       timer_settime pthread_create pthread_attr_init pthread_attr_setdetachstate
@@ -82,4 +75,4 @@ add_library( iotplatform
 set_target_properties( iotplatform PROPERTIES VERSION ${PROJECT_VERSION} )
 
 # Link required libraries.
-target_link_libraries( iotplatform Threads::Threads rt ${TLS_LIBRARY_LINKER_FLAG} )
+target_link_libraries( iotplatform Threads::Threads ${TLS_LIBRARY_LINKER_FLAG} )
diff --git a/platform/source/posix/iot_network_openssl.c b/platform/source/posix/iot_network_openssl.c
index da804ef..44d6af5 100644
--- a/platform/source/posix/iot_network_openssl.c
+++ b/platform/source/posix/iot_network_openssl.c
@@ -36,7 +36,7 @@
 
 /* POSIX includes. */
 #include <errno.h>
-#include <poll.h>
+#include <sys/poll.h>
 #include <pthread.h>
 #include <signal.h>
 #include <sys/ioctl.h>
@@ -44,6 +44,9 @@
 /* Sockets includes. */
 #include <netdb.h>
 #include <sys/socket.h>
+#include <netinet/in.h>
+#include <netinet/ip.h>
+#include <arpa/inet.h>
 
 /* OpenSSL includes. */
 #include <openssl/err.h>
