diff --git a/lib/source/common/iot_taskpool.c b/lib/source/common/iot_taskpool.c
index 031e7d2..7290573 100644
--- a/lib/source/common/iot_taskpool.c
+++ b/lib/source/common/iot_taskpool.c
@@ -61,7 +61,7 @@
 /**
  * @brief Maximum semaphore value for wait operations.
  */
-#define TASKPOOL_MAX_SEM_VALUE              0xFFFF
+#define TASKPOOL_MAX_SEM_VALUE              0x7FFF
 
 /**
  * @brief Reschedule delay in milliseconds for deferred jobs.
